File: Data Structures & Algorithms/sliding-window-maximum/submission-2.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        outs = []
        p1=0
        p2=p1+k
        while p2<=len(nums):
            outs.append(max(nums[p1:p2]))
            p1+=1
            p2+=1
        return outs

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> count_dict;
        for (int i = 0; i < nums.size(); i++) {
            if (count_dict.find(nums[i]) == count_dict.end()) {
                count_dict[nums[i]] = 1;
            } else {
                count_dict[nums[i]] += 1;
                if (count_dict[nums[i]] > 1) {
                    return true;
                }
            }
        }
        return false;
    }
};